src/compiler/utilities.ts

```diff
@@ -2414,6 +2414,27 @@ namespace ts {
         return output;
     }
 
+    export const stringify: (value: any) => string = JSON && JSON.stringify ? JSON.stringify : function stringify(value: any): string {

```

Comment:
I added a check that is only verified when the current assertion level is Aggressive or higher (although this only ever happens if we explicitly set this value as its a constant). I want to avoid introducing a closure for `reduceRight` just to keep track of cycles.